### PR TITLE
Add mathjax to properly render mathjax content

### LINKED ISSUE #9: Apply proper MathJax
Once #7 is implemented, we need to properly install MathJax, i.e. proper version and configuration.

Both seems to be present in page content on the API, in the `tail` attribute. 

```
{
...,
"tail": "<!-- Google tag (gtag.js) -->\r\n<script async=\"async\" src=\"https://www.googletagmanager.com/gtag/js?id=G-QBY3PCNWX9\"></script>\r\n<script>/*<![CDATA[*/\r\n  window.dataLayer = window.dataLayer || [];\r\n  function gtag(){dataLayer.push(arguments);}\r\n  gtag('js', new Date());\r\n\r\n  gtag('config', 'G-QBY3PCNWX9');\r\n/*]]>*/</script>\r\n\r\n<script type=\"text/x-mathjax-config\">/*<![CDATA[*/\r\n  MathJax.Ajax.config.path[\"mhchem\"] =\r\n  \"https://cdnjs.cloudflare.com/ajax/libs/mathjax-mhchem/3.3.2\";\r\n  MathJax.Hub.Config({ jax: [\"input/TeX\",\"input/MathML\",\"output/SVG\"],\r\n  extensions: [\"tex2jax.js\",\"mml2jax.js\",\"MathMenu.js\",\"MathZoom.js\"],\r\n  TeX: {\r\n    extensions: [\"autobold.js\",\"mhchem.js\",\"color.js\",\"cancel.js\", \"AMSmath.js\",\"AMSsymbols.js\",\"noErrors.js\",\"noUndefined.js\"]\r\n  },\r\n    \"HTML-CSS\": { linebreaks: { automatic: true , width: \"90%\"}, scale: 85, mtextFontInherit: false},\r\nmenuSettings: { zscale: \"150%\", zoom: \"Double-Click\" },\r\n         SVG: { linebreaks: { automatic: true } }});\r\n/*]]>*/</script>\r\n\r\n<script type=\"text/javascript\" async=\"true\" src=\"https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS_HTML\"></script>"
}
```

So MathJax is not something supposed to be applied unconditionally on all pages (especially if we want the scraper to stay generic). We could theoretically have different MathJax version on same website.

It is however clear that we have other stuff on this attribute (here Google Tag). Maybe we should just clean / keep only mathjax related stuff.

### PR DESCRIPTION
**Pull request by benoit74**
Fix #9 

Changes:
- install MathJax from sources on scraper side (failed to work properly on JS side since we want to include it dynamically)
- setup MathJax based on libretexts.org configuration / needs
- add logic to reload MathJax everytime we change page since we need to compute current page `front`, used to properly number equations / figures / ...
- reorder processing steps to do "simple and fast" things first
- adapt `CONTRIBUTING.md` 
- also add page title to display

### CODE REVIEW

File: zimui/src/directives/mathjax.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          zscale: '150%'
+        }
+      }
+    }
+  }
+  const script = document.createElement('script', {
+    id: 'mathjax-script'
+  } as ElementCreationOptions)
+  script.src = './mathjax/es5/tex-svg.js'
+  script.async = false
```
Comment (rgaudin) [inline zimui/src/directives/mathjax.ts:74]: This can be async without any issue btw.

**Review by rgaudin — APPROVED** [review]
I find this a but ugly as `updated` is completely outside our control and could be called many times.
Given we want to update the macro when page change, it makes more sens to me to tie this to a page change and not a DOM mutation observer.

It works without noticeable impact though so if you feel that's more maintainable this way, it's fine by me.

File: zimui/src/directives/mathjax.ts (hunk 1 above)
Comment (benoit74) [reply zimui/src/directives/mathjax.ts:74]: OK, moved to async

**Comment by benoit74** [comment]
For the record, I've moved from a directive to a service, since this indeed seems more appropriate / simple (and probably more maintainable).

**Pull request merged**